### ISSUE TITLE
RN 0.46 - Realm has no setter or ivar for it's bridge

### ISSUE THREAD

**Issue by ovr** [realm/realm-js#1121]
When I am using non remote js debug mode, I am getting

![image](https://user-images.githubusercontent.com/572096/27895087-1dbc8118-624b-11e7-99c1-daa608fee154.png)

## Version of Realm and Tooling

- Realm JS SDK Version: 1.8.3
- Node or React Native: 0.46.0 
- Client OS & Version: iOS

There was similar error but for 0.45 https://github.com/realm/realm-js/issues/1049

**Comment by ovr** [comment]
Exception after calling `executeBlockOnJavaScriptThread` on `bridge`

![image](https://user-images.githubusercontent.com/572096/27895260-7a2a9e02-624c-11e7-98ca-9ce6ddfe0825.png)

**Comment by compojoom** [comment]
I have the same error with disabled JS debug. When js debug is on - the error is not there. (in my case it is 0.46.0rc2

**Comment by thatjuan** [comment]
+1

**Comment by ovr** [comment]
Yeap, It's exception

![image](https://user-images.githubusercontent.com/572096/27899019-79f1b4fe-6263-11e7-9b18-f5dae627d977.png)

**Comment by ovr** [comment]
So, I started to use `ensureOnJavaScriptThread`, going to prepare PR

**Comment by ovr** [comment]
@javache Hey bro! You introduced changes inside `react-native` repository https://github.com/facebook/react-native/commit/8525aac769a08ca63fd932ac19c6bbbee81d39fb#diff-a2a67635fffd7b690d14dc17ae563a71

So, main question can you give a good advice how to check method exists compatiblity for RN 0.46+

Currently, inside https://github.com/ovr/realm-js/commit/71614726c110512499a4244eef0b14f7b439afb6

I am using

```objc
        Class RCTCxxBridgeClass = NSClassFromString(@"RCTCxxBridge");
        if ([RCTCxxBridgeClass instancesRespondToSelector:@selector(ensureOnJavaScriptThread:)]) {
            [bridge ensureOnJavaScriptThread:bridgeInitializer];
        } else {
            [bridge executeBlockOnJavaScriptThread:bridgeInitializer];
        }
```

Thank you 🍺 🍻

**Comment by javache** [comment]
If you're going to be using private API's (which is not supported, and will likely break again in future releases), use `dispatchBlock:block queue:RCTJSThread`, which is implemented in both the old and the new bridge.

**Comment by ovr** [comment]
@javache Thank you

**Comment by fealebenpae** [comment]
Fixed in #1122
